<?xml version="1.0"?>
<launch>
	<arg name="teleop" default="joy" />
   <arg name="filter" default="off"/>

   <include file="$(find cyberpod_sim_ros)/launch/manual_teleop.launch" if="$(eval teleop == 'joy')"/>

   <include file="$(find cyberpod_sim_ros)/launch/rviz.launch"/>

	<group ns="cyberpod_sim_ros">
		
		<node name="integrator" pkg="cyberpod_sim_ros" type="integrator_node" output="screen">
			<param name="dt" value="0.001" />
			<param name="umax" value="20.0" />
			<rosparam param="IC">[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]</rosparam>
		</node>

		<node name="safety_filter" pkg="cyberpod_sim_ros" type="safety_filter_node" output="screen">
			<param name="pass_through" value="0" if="$(eval filter == 'on')"/>
			<param name="pass_through" value="1" if="$(eval filter == 'off')"/>
			<param name="integration_dt" value="0.01"/>
			<param name="backup_Tmax" value="1.3"/>
		</node>

		<node name="controller" pkg="cyberpod_sim_ros" type="controller_node" output="screen">
			<param name="offset_angle" value="0.1383244254"/>
			<rosparam param="gains">[44.7214, 0., 0., 44.6528, 0., 150.1612, 37.6492]</rosparam>
		</node>

	</group>
</launch>